components/raftstore/src/store/snapshot_backup.rs

```diff
@@ -0,0 +1,388 @@
+    fn pre_propose_query(
+        &self,
+        cx: &mut crate::coprocessor::ObserverContext<'_>,
+        reqs: &mut Vec<kvproto::raft_cmdpb::Request>,
+    ) -> crate::coprocessor::Result<()> {
+        if self.allowed() {
+            return Ok(());
+        }
+        for req in reqs {
+            if req.has_ingest_sst() {
+                // Note: this will reject the batch of commands, which isn't so effective.
+                // But we cannot reject proposing a subset of command for now...
+                cx.bypass = true;
+                metrics::SNAP_BR_SUSPEND_COMMAND_TYPE
+                    .with_label_values(&["Ingest"])
+                    .inc();
+                return Err(self.reject());
+            }
+        }
+        Ok(())
+    }
+}
+
+impl AdminObserver for Arc<PrepareDiskSnapObserver> {
+    fn pre_propose_admin(
+        &self,
+        _: &mut crate::coprocessor::ObserverContext<'_>,
+        admin: &mut kvproto::raft_cmdpb::AdminRequest,
+    ) -> crate::coprocessor::Result<()> {
+        if self.allowed() {
+            return Ok(());
+        }
+        if matches!(
+            admin.get_cmd_type(),
+            AdminCmdType::Split |
+            AdminCmdType::BatchSplit |
+            // We disable `CompactLog` here because if the log get truncated, 
+            // we may take a long time to send snapshots during restoring.
+            AdminCmdType::CompactLog |
+            // We will allow `Commit/RollbackMerge` here because the 
+            // `wait_pending_admin` will wait until the merge get finished.
+            // If we reject them, they won't be able to see the merge get finished.
+            // And will finally time out.
+            AdminCmdType::PrepareMerge |
+            AdminCmdType::ChangePeer |
+            AdminCmdType::ChangePeerV2
+        ) {
+            metrics::SNAP_BR_SUSPEND_COMMAND_TYPE
+                .with_label_values(&[&format!("{:?}", admin.get_cmd_type())])
+                .inc();
+            return Err(self.reject());
+        }
+        Ok(())
+    }
+
+    fn pre_transfer_leader(
+        &self,
+        _ctx: &mut crate::coprocessor::ObserverContext<'_>,
+        _tr: &kvproto::raft_cmdpb::TransferLeaderRequest,
+    ) -> crate::coprocessor::Result<()> {
+        if self.allowed() {
+            return Ok(());
+        }
+        metrics::SNAP_BR_SUSPEND_COMMAND_TYPE
+            .with_label_values(&["TransferLeader"])
+            .inc();
+        Err(self.reject())
+    }
+}
+
+#[derive(Debug)]
+struct SyncerCore {
+    report_id: u64,
+    feedback: Option<oneshot::Sender<SyncReport>>,
+}
+
+#[derive(Debug, PartialEq)]
+pub struct SyncReport {
+    pub report_id: u64,
+    pub aborted: Option<AbortReason>,
+}
+
+impl SyncerCore {
+    fn new(report_id: u64, feedback: oneshot::Sender<SyncReport>) -> Self {
+        Self {
+            report_id,
+            feedback: Some(feedback),
+        }
+    }
+
+    fn aborted(&self) -> bool {
```

Comment:
A method/function that returns a boolean should be named as a question, eg., `is_aborted`?